test/images/agnhost/entrypoint-tester/ep.go

```diff
@@ -14,16 +14,25 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package main
+package entrypoint
 
 import (
 	"fmt"
 	"os"
+
+	"github.com/spf13/cobra"
 )
 
+// CmdEntrypointTester is used by agnhost Cobra.
+var CmdEntrypointTester = &cobra.Command{
+	Use:   "entrypoint-tester",
+	Short: "Prints the args it's passed and exits",
+	Long:  "Prints the args it's passed and exits.",
+	Run:   main,
+}
+
 // This program prints the arguments it's passed and exits.
-func main() {
-	args := os.Args
-	fmt.Printf("%v\n", args)
+func main(cmd *cobra.Command, args []string) {
+	fmt.Printf("%v\n", os.Args)
```

Comment:
should this be `args`?

Reply:
After reviewing how this is used (ref:https://github.com/kubernetes/kubernetes/pull/79238#pullrequestreview-254347384) 

I think it's correct and intentional that this uses `os.Args`; maybe add a comment explaining as much?  Or something like change `main` to not take `args`?

Reply:
Yep, that is correct, and I will add a comment why it is so.

Basically, the parameter ``args`` is given by Cobra, and I cannot change that, it won't be compilable. Basically, ``args`` will contain the subcommand's arguments (``./agnhost subcommand args here``), while the tests are also verifying the entrypoint name (``os.Args[0]``) (there are tests which are overriding the entrypoint).